src/Control/SudoModeController.php

```diff
@@ -0,0 +1,172 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\SecurityExtensions\Control;
+
+use SilverStripe\Admin\LeftAndMain;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Authenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\SecurityToken;
+use SilverStripe\SecurityExtensions\Service\SudoModeServiceInterface;
+
+/**
+ * Responsible for checking and verifying whether sudo mode is enabled
+ */
+class SudoModeController extends LeftAndMain
+{
+    private static $url_segment = 'sudomode';
+
+    private static $ignore_menuitem = true;
+
+    private static $allowed_actions = [
+        'check',
+        'activate',
+    ];
+
+    private static $dependencies = [
+        'SudoModeService' => '%$' . SudoModeServiceInterface::class,
+    ];
+
+    /**
+     * A user help documentation link to find out more about sudo mode
+     *
+     * @config
+     * @var string
+     */
+    private static $help_link = 'http://google.com';
+
+    /**
+     * @var SudoModeServiceInterface
+     */
+    private $sudoModeService;
+
+    public function getClientConfig()
+    {
+        /** @var HTTPRequest $request */
+        $request = Injector::inst()->get(HTTPRequest::class);
+
+        return array_merge_recursive(parent::getClientConfig(), [
+            'endpoints' => [
+                'activate' => $this->Link('activate'),
+            ],
+            'sudoModeActive' => $this->getSudoModeService()->check($request->getSession()),
+            'helpLink' => $this->config()->get('help_link'),
+        ]);
+    }
+
+    /**
+     * Checks whether sudo mode is active for the current user
+     *
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     */
+    public function check(HTTPRequest $request): HTTPResponse
+    {
+        return $this->jsonResponse([
+            'active' => $this->getSudoModeService()->check($request->getSession()),
+        ]);
+    }
+
+    /**
+     * After validating the request data including password against the current member, activate sudo mode
+     * for the current member.
+     *
+     * @param HTTPRequest $request
+     * @return HTTPResponse
+     * @throws HTTPResponse_Exception If the request was not made with POST
+     */
+    public function activate(HTTPRequest $request): HTTPResponse
+    {
+        if (!$request->isPOST()) {
+            return $this->httpError(404);
+        }
+
+        if (!SecurityToken::inst()->checkRequest($request)) {
+            return $this->jsonResponse([
+                'result' => false,
+                'message' => _t(__CLASS__ . '.TIMEOUT', 'Session timed out, please try again.'),
```

Comment:
I think this would make more sense to say "refresh and try again". As it's a JSON response it'll likely be used with AJAX.